lib/Sema/TypeCheckStmt.cpp

```diff
@@ -2221,6 +2221,13 @@ static bool checkSuperInit(ConstructorDecl *fromCtor,
       fromCtor->diagnose(diag::availability_unavailable_implicit_init,
                          ctor, superclassDecl->getName());
     }
+    
+    // Not allowed to implicitly generate a super.init() call if it's
+    // a throwing initializer while the one being typechecked is not.
+    if ((fromCtor->hasThrows() == false) && ctor->hasThrows()) {
```

Comment:
Stylistically we prefer `!x` over `x == false`

Reply:
Improved!